### DISCUSSION TITLE
CPEX 0046 - New particle extension proposal

### DISCUSSION THREAD

**Discussion by AlexandreMinotCSI** [CGNS/CGNS#766] [CGNStalk]
Dear CGNS community,

We would like to propose a new extension of the CGNS standard to include particle data. Please find attached a document that details our proposal. We were not able to create a new Jira issue, so we are posting here only. We also have code changes to the SIDS, FMM and MLL. We cannot figure out how to push those changes back to the Github. The server is rejecting our credentials. Please instruct on the best way to contribute code to those two projects.

Please let us know what you think so we may adjust our proposal.

[CPEX0046-v2.pdf](https://github.com/user-attachments/files/15915317/CPEX0046-v2.pdf)

**Comment by jrwrigh** [comment]
> Please instruct on the best way to contribute code to those two projects.

The standard way to do this would be to create a fork of the repo and push code onto that. GitHub can then allow you to create a PR from that fork into the original repository itself. [See GirHub documentation.](https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project)

**Comment by AlexandreMinotCSI** [comment]
Thanks for your help. Here are the changes:
SIDS, FMM, and MLL doc: https://github.com/CGNS/cgns.github.io/pull/26
MLL code: https://github.com/CGNS/CGNS/pull/767

**Comment by brtnfld** [comment]
@AlexandreMinotCSI, we have a CGNS steering committee meeting on July 9th, where we will review your CPEX. Would you be available to attend that meeting?

**Reply by AlexandreMinotCSI** [reply]
Thanks you for the quick response. Yes we should be able to attend. Here is my email: alexandre.minot@convergecfd.com

**Comment by AlexandreMinotCSI** [comment]
Proposal revision 1: added authors' email adresses
[CPEX0046-v2-rev1.pdf](https://github.com/user-attachments/files/15930721/CPEX0046-v2-rev1.pdf)

**Comment by brtnfld** [comment]
48 is already taken. Since this is a rewrite/completion of the effort started by Thomas, it is okay to number it 46. But you might want to mention the v1 CPEX or reference it when needed.

**Reply by AlexandreMinotCSI** [reply]
Sounds good, I'll stick with revision 1 then.

**Reply by MicK7** [reply]
Sorry for the mix-up

**Comment by brtnfld** [comment]
Created issue #772  for further discussion and updates.
